### ISSUE TITLE
allowing Strict-Transport-Security (HSTS)

### ISSUE THREAD

**Issue by pointhi** [searx/searx#62]
For security reasons, an option to enable HSTS would be a good idea.

**Comment by wiewo** [comment]
had the same problem with apache+uWSGI. Adding the following line to searx.ini solved it:

```
route-if = equal:${HTTPS};on addheader:Strict-Transport-Security: max-age=31536000
```

**Comment by pointhi** [comment]
I tried it, but it doesn't worked for me. I'm using uwsgi with the version: `1.2.3-debian`. Probably it is not supported in this version.

**Comment by wiewo** [comment]
possible, I'm on debian jessie with uwsgi version 2.0.7. Maybe it's also possible to force apache to add its header data?

**Comment by xinomilo** [comment]
i dont use apache, but can you try these instructions : https://raymii.org/s/tutorials/HTTP_Strict_Transport_Security_for_Apache_NGINX_and_Lighttpd.html , on the apache searx config file?

**Comment by pointhi** [comment]
Sorry, but the configuration:

```
Header always set Strict-Transport-Security "max-age=15768000"
```

as well as

```
Header always add Strict-Transport-Security "max-age=15768000"
```

doesn't work.

**Explanation:**

https://stackoverflow.com/questions/25781833/can-mod-headers-change-headers-generated-by-uwsgi

**Comment by pointhi** [comment]
Possible solution, add response header inside searx:

https://stackoverflow.com/questions/25860304/how-do-i-set-response-headers-in-flask

**Comment by xinomilo** [comment]
just found this, dont know if it helps.: 
https://uwsgi-docs.readthedocs.org/en/latest/Snippets.html#force-https

**Comment by pointhi** [comment]
my instance is already forced to https using apache. I want to use HSTS mainly to secure my instance again downgrade attacks.

**Comment by xinomilo** [comment]
i dont think that affects you in the scenario provided in the link. shouldn't be  different

**Comment by asciimoo** [comment]
This feature is included in #715
